mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java

```diff
@@ -858,13 +861,28 @@ public boolean onPrepareOptionsMenu(Menu menu) {
         Log.d(TAG, "onPrepareOptionsMenu()");
         MenuItem voiceRecognitionItem = menu.findItem(R.id.mainmenu_voice_recognition);
         @ColorInt int iconColor = ContextCompat.getColor(this, R.color.light);
-        voiceRecognitionItem.setVisible(mConnection != null
-                && SpeechRecognizer.isRecognitionAvailable(this));
-        manageVoiceRecognitionShortcut(SpeechRecognizer.isRecognitionAvailable(this));
+        voiceRecognitionItem.setVisible(mConnection != null);
         voiceRecognitionItem.getIcon().setColorFilter(iconColor, PorterDuff.Mode.SRC_IN);
+
+        boolean isSpeechRecognizerAvailable = SpeechRecognizer.isRecognitionAvailable(this);
+        manageVoiceRecognitionShortcut(isSpeechRecognizerAvailable);
+        setVoiceWidgetComponentEnabledSetting(VoiceWidget.class.getCanonicalName(),
+                isSpeechRecognizerAvailable);
+        setVoiceWidgetComponentEnabledSetting(VoiceWidgetWithIcon.class.getCanonicalName(),
+                isSpeechRecognizerAvailable);
```

Comment:
Doing this every time when populating the menu seems a bit excessive. I'd suggest doing this in an AsyncTask fired from onCreate() instead.

Reply:
Done